fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -550,8 +552,8 @@ struct DDQueueData {
 	            FutureStream<RelocateShard> input,
 	            PromiseStream<GetMetricsRequest> getShardMetrics,
 	            PromiseStream<GetTopKMetricsRequest> getTopKMetrics)
-	  : distributorId(mid), lock(lock), cx(cx), teamCollections(teamCollections), shardsAffectedByTeamFailure(sABTF),
-	    getAverageShardBytes(getAverageShardBytes),
+	  : distributorId(mid), lock(lock), cx(cx), txnProcessor(new DDTxnProcessor(cx)), teamCollections(teamCollections),
```

Comment:
nit: is there some reason we are not using a smart pointer for `txnProcessor`?

Reply:
Actually, this constructor should change to accept an `IDDTxnProcessor` obj as an argument for dependency inverse purpose once I move all the needed transaction code into `DDTxnProcessor` in the future, and at that time `DDQueue, DDTracker, DDTeamCollection` will share the same IDDTxnProcessor obj.
For the purpose to keep this PR small, I just want to make small changes to the current code.

Reply:
My bad, I missed the std::shared_ptr, :).